client/components/plans/plan-price/index.jsx

```diff
@@ -62,13 +62,30 @@ module.exports = React.createClass( {
 			periodLabel = hasDiscount ? this.translate( 'due today when you upgrade' ) : plan.bill_period_label
 		}
 
-		return (
-			<div className={ hasDiscount ? "plan-price plan-price__discount" : "plan-price" }>
-				<span>{ this.getPrice() }</span>
-				<small className="plan-price__billing-period">
-					{ periodLabel }
-				</small>
-			</div>
-		);
+		if ( site && site.jetpack ) {
```

Comment:
What about creating a new component specifically for Jetpack plan prices? That way we don't need this big if/else.